ebean-datasource/src/main/java/io/ebean/datasource/pool/PooledConnection.java

```diff
@@ -426,16 +470,17 @@ public void close() throws SQLException {
         resetIsolationReadOnlyRequired = false;
       }
 
-      if (resetSchema) {
+      if (schemaState == SCHEMA_CATALOG_CHANGED) {
         connection.setSchema(originalSchema);
-        currentSchema = null;
-        resetSchema = false;
+        // we can use original value for cache computation from now on
+        cacheKeySchema = originalSchema;
```

Comment:
Possible bug: We should not set `currentSchema` respectively `cacheKey` to `null`

Reply:
Yes, good !!